pkg/webhook/v1beta1/experiment/validator/validator.go

```diff
@@ -172,6 +175,21 @@ func (g *DefaultValidator) validateAlgorithm(ag *commonapiv1beta1.AlgorithmSpec)
 	return nil
 }
 
+func (g *DefaultValidator) validateEarlyStopping(es *commonapiv1beta1.EarlyStoppingSpec) error {
+	if es == nil {
+		return nil
+	}
+	if es.AlgorithmName == "" {
+		return fmt.Errorf("no spec.earlyStopping.algorithm.algorithmName specified")
+	}
+
+	if _, err := g.GetEarlyStoppingConfigData(es); err != nil {
+		return fmt.Errorf("unable to get EarlyStopping config data for algorithm %s: %v", es.AlgorithmName, err)
+	}
```

Comment:
Regarding to my comment: https://github.com/kubeflow/katib/issues/1680#issuecomment-928164345.
I was thinking about Webhook Validation only for the AlgorithmName, and for the check if EarlyStopping exists in Katib Config. Exactly what you are doing here, similar to [`ValidateAlgorithm`](https://github.com/kubeflow/katib/blob/master/pkg/webhook/v1beta1/experiment/validator/validator.go#L159-L173).

Sorry if I confused you.